runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -128,10 +139,19 @@ public TableRegistry(CorfuRuntime runtime) {
                 .setSerializer(this.protobufSerializer)
                 .open();
 
+        this.protobufDescriptorTable = this.runtime.getObjectsView().build()
```

Comment:
I'm a bit confused, if the problem is memory limitations, aren't we opening a new table which contains the data already in registryTable? so it's like duplication between them? shouldn't the table registry only contain the type and not the definitions, maybe a mapping to the file name.

Reply:
That's exactly what it is - We don't add the descriptors in the table registry, only the file names. the actual descriptors which are the memory hogs are de-duplicated in the protobuf descriptor table.